syntax = "proto3";

package demo;

service DemoService {
  rpc UnaryCall (RequestMessage) returns (ResponseMessage);
  rpc ServerStreamingCall (RequestMessage) returns (stream ResponseMessage);
  rpc ClientStreamingCall (stream RequestMessage) returns (ResponseMessage);
  rpc BidiStreamingCall (stream RequestMessage) returns (stream ResponseMessage);
}

message RequestMessage {
  string message = 1;
}

message ResponseMessage {
  string message = 1;
}
